prereise/gather/hiflddata/data_trans.py

```diff
@@ -0,0 +1,645 @@
+    """Clean data; remove substations which are outside the United States or not available.
+
+    :param str E_csv: path of the HIFLD substation csv file
+    :param dict zone_dic: zone dict as returned by :func:`get_Zone`
+    :return: (*pandas.DataFrame*) -- a pandas Dataframe storing the substations after dropping the invalid ones.
+    """
+
+    csv_data = pd.read_csv(E_csv)
+    Num_sub = len(csv_data)
+    row_indexs = []
+    for i in range(Num_sub):
+        if (
+            (csv_data["STATE"][i] not in zone_dic)
+            or (csv_data["STATUS"][i] != "IN SERVICE")
+            or (csv_data["LINES"][i] == 0)
+        ):
+            row_indexs.append(i)
+    clean_data = csv_data.drop(labels=row_indexs)
+    return clean_data
+
+
+def Neighbors(sub_by_coord_dict, sub_name_dict, raw_lines):
+    """Create dict to store the neighbors of each substation based on transmission topology geojson from HIFLD
+
+    :param dict sub_by_coord_dict: dict mapping the (x, y) to a substation
+    :param dict sub_name_dict: dict mapping the substation name to a substation
+    :return: (*list*) -- lines, a list of lines after the unambiguous substation name cleanup.
+    :return: (*list*) -- nodes, a list of substations after the unambiguous substation name cleanup.
+    :return: (*dict*) -- N_dict, a dict mapping the substation to its neighbor list.
+    """
+
+    N_dict = {}
+    nodes = []
+    lines = []
+    missinglines = []
+    if not os.path.isfile("data/unzip/Electric_Power_Transmission_Lines.geojson"):
+        with zipfile.ZipFile(
+            "data/Electric_Power_Transmission_Lines.geojson.zip", "r"
+        ) as zip_ref:
+            zip_ref.extractall("data/unzip")
+    with open(
+        "data/unzip/Electric_Power_Transmission_Lines.geojson", "r", encoding="utf8"
+    ) as fp:
+        data = json.load(fp)
+    for i in range(len(data["features"])):
+        line = data["features"][i]
+        line_id = line["properties"]["ID"]
+        line_type = line["properties"]["TYPE"]  # e.g. "AC; OVERHEAD"
+        if (
+            line["properties"]["SUB_1"] == "NOT AVAILABLE"
+            or line["properties"]["SUB_2"] == "NOT AVAILABLE"
+        ):
+            # quite a few of NOT AVAILABLE scenarios in transmission rawdata
+            print("INFO: sub1 or sub2 NOT AVAILABLE for transmission line: ", line)
+            missinglines.append(line)
+            continue
+
+        if (
+            line["properties"]["SUB_1"] not in sub_name_dict
+            or line["properties"]["SUB_2"] not in sub_name_dict
+        ):
+            # e.g., ID 201978 with SUB2 UNKNOWN202219, never shown in substation rawdata
+            print(
+                "INFO: sub1 or sub2 name not identified for transmission line: ", line
+            )
+            missinglines.append(line)
+            continue
+
+        start_coord = (
+            line["geometry"]["coordinates"][0][0][1],
+            line["geometry"]["coordinates"][0][0][0],
```

Comment:
What do each of these axes represent?

Reply:
line["geometry"]["coordinates"][0][0] is the start point (substation) for this transmission line, with (line["geometry"]["coordinates"][0][0][0], line["geometry"]["coordinates"][0][0][1]) being the (latitude, longitude) for this substation.

Reply:
What do the first and second zeros in `[0][0][0]`/`[0][0][1]` represent? What are these layers of indexing/slicing? A comment in the source code would make this more clear.

Reply:
Ok, I will add more detailed comments in next revision.